### PR TITLE
[BACKLOG-24517] Adds explicit version numbers on exported packages. A…

### PR DESCRIPTION
**Pull request by lucboudreau**
…lso fixes some conversion mistakes.

### CODE REVIEW

**Comment by lucboudreau** [comment]
@pentaho/moseisley For review.

File: common-services-api/hbase/pom.xml [hunk 1]
```diff
@@ -67,7 +67,8 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-            <Export-Package>org.pentaho.bigdata.api.hbase,org.pentaho.bigdata.api.hbase.*</Export-Package>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
+            <Export-Package>org.pentaho.bigdata.api.hbase;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (jgminder) [inline common-services-api/hbase/pom.xml]: Missing wildcard

File: shims/hdp26/hbase/pom.xml [hunk 2]
```diff
@@ -237,14 +237,15 @@
...
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
             <DynamicImport-Package>*</DynamicImport-Package>
             <Import-Package>
               org.pentaho.hadoop.shim.api.*;version=${pentaho-osgi-bundles.version},
               org.pentaho.hadoop.shim.spi.*;version=${pentaho-osgi-bundles.version},
               io.netty.channel;version=${bundle.export.version},
               org.apache.hadoop.conf.*;version=${bundle.export.version}
             </Import-Package>
-            <_exportcontents>org.pentaho.hbase.shim.common,org.apache.hadoop.hbase.*;version=${bundle.export.version}</_exportcontents>
+            <_exportcontents>org.pentaho.hbase.shim.common;version=${bundle.export.version}</_exportcontents>
```
Comment (jgminder) [inline shims/hdp26/hbase/pom.xml]: Why did we leave out the org.apache.hadoop.hbase.* packages? If we're including hbase jars in this uber jar, we need to export them.

Comment (lucboudreau) [reply shims/hdp26/hbase/pom.xml]: None of the other shim's hbase modules export this package. It also does not exist in this project. I removed it for consistency. I doubt it'll break anything.

File: common-services-api/hbase/pom.xml (hunk 1 above)
Comment (lucboudreau) [reply common-services-api/hbase/pom.xml]: Yep. This should be a wildcard. I'll fix it.

File: shims/hdp26/hive/pom.xml [hunk 3]
```diff
@@ -192,14 +192,13 @@
...
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
             <DynamicImport-Package>*</DynamicImport-Package>
             <Import-Package>
               org.pentaho.hadoop.shim.api.*,
               org.pentaho.hadoop.shim.spi.*,
             </Import-Package>
-            <_exportcontents>
-              org.apache.hadoop.hive.*;version=${bundle.export.version}
-            </_exportcontents>
+            <_exportcontents>org.apache.hadoop.hive;version=${bundle.export.version}</_exportcontents>
```
Comment (jgminder) [inline shims/hdp26/hive/pom.xml]: Has hive been tested with this change? We left the wildcard off and it's fine to do as long as all the hive packages are covered by this.

File: shims/hdp25/hive/pom.xml [hunk 4]
```diff
@@ -192,14 +192,13 @@
...
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
             <DynamicImport-Package>*</DynamicImport-Package>
             <Import-Package>
               org.pentaho.hadoop.shim.api.*,
               org.pentaho.hadoop.shim.spi.*,
             </Import-Package>
-            <_exportcontents>
-              org.apache.hadoop.hive.*;version=${bundle.export.version}
-            </_exportcontents>
+            <_exportcontents>org.apache.hadoop.hive;version=${bundle.export.version}</_exportcontents>
```
Comment (jgminder) [inline shims/hdp25/hive/pom.xml]: See the comment regarding hdp26.

File: shims/hdp26/hive/pom.xml (hunk 3 above)
Comment (lucboudreau) [reply shims/hdp26/hive/pom.xml]: You are correct. I double checked and this should be a wildcard too.

File: shims/hdp25/hive/pom.xml (hunk 4 above)
Comment (lucboudreau) [reply shims/hdp25/hive/pom.xml]: You are correct. I double checked and this should be a wildcard too.

**Comment by lucboudreau** [comment]
@jgminder PR was updated.

File: common-services-api/format/pom.xml [hunk 5]
```diff
@@ -32,7 +32,9 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
             <Import-Package>org.pentaho.hadoop.shim.api.format.*,!org.pentaho.hbase.shim.api,!org.pentaho.hbase.shim.spi,!org.apache.hadoop.*,!org.pentaho.di.core.hadoop.*,!org.pentaho.hadoop.*,!org.pentaho.yarn.*,!org.apache.log4j,*</Import-Package>
+            <Export-Package>org.pentaho.bigdata.api.format;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (lucboudreau) [inline common-services-api/format/pom.xml]: There's only one package in this project. No need for the wildcard.

File: common-services-api/jdbc/pom.xml [hunk 6]
```diff
@@ -56,6 +56,8 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
+            <Export-Package>org.pentaho.big.data.api.jdbc;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (lucboudreau) [inline common-services-api/jdbc/pom.xml]: This is properly split into interface and implementation. We only need to export the interface package.

File: common-services-api/mapreduce/pom.xml [hunk 7]
```diff
@@ -47,6 +47,8 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
+            <Export-Package>org.pentaho.bigdata.api.mapreduce;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (lucboudreau) [inline common-services-api/mapreduce/pom.xml]: There's only one package in this project. No need for the wildcard.

File: common-services-api/oozie/pom.xml [hunk 8]
```diff
@@ -31,7 +31,8 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-            <Export-Package>org.pentaho.bigdata.api.oozie</Export-Package>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
+            <Export-Package>org.pentaho.bigdata.api.oozie;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (lucboudreau) [inline common-services-api/oozie/pom.xml]: There's only one package in this project. No need for the wildcard.

File: common-services-api/pig/pom.xml [hunk 9]
```diff
@@ -53,7 +53,8 @@
         <configuration>
           <instructions>
             <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-            <Export-Package>org.pentaho.bigdata.api.pig,org.pentaho.bigdata.api.pig.impl</Export-Package>
+            <Bundle-Version>${pentaho-osgi-bundles.version}</Bundle-Version>
+            <Export-Package>org.pentaho.bigdata.api.pig.*;version=${pentaho-osgi-bundles.version}</Export-Package>
```
Comment (lucboudreau) [inline common-services-api/pig/pom.xml]: This is properly split into interface and implementation. We only need to export the interface package.

**Pull request merged**